### PR TITLE
Combine all the recent work

### PR DESCRIPTION
**Pull request by dirkhh**
### Describe the pull request:

- [x] Bug fix
- [x] Functional change
- [ ] New feature
- [x] Code cleanup
- [x] Build system change
- [ ] Documentation change
- [ ] Language translation

### Pull request long description:

This is a combination of #2302 and all the work I had done to deal with the Android UI issues, plus a number of small build system changes and other random cleanup
Not very structured, but it's a very close match for the latest Android beta; this one has one merge commit left that I had flattened in my internal branch that I used to create the Android binary - that's why the version numbers don't match - but having that merge seemed to make things easier to figure out in the git history.

### Changes made:

1) @bstoeger 's fixes for the way we deal with the table of downloaded dives
2) some very hacky QML code that tries to be clever about the spurious screen size change that we get on some devices... I think it is related to the way I change the font and related information in certain situations, but without that Subsurface-mobile is unusable on certain small devices with large default fonts
3) a few random cleanups and build system modifications
4) a new version number for Subsurface-mobile

### Related issues:

#2292
#2295 
#2301 

### Mentions:

@bstoeger I'd appreciate a review :-/

### CODE REVIEW

**Comment by dirkhh** [comment]
I think the only commit that isn't signed off is the merge commit. Hrmpf.

**Comment by bstoeger** [comment]
I'll have a look later today.
Just note that the downloadTable thing was written before we realized that model-reset may cause a rendering of the dive-detail page for *all* dives. So this might need some tuning. Nevertheless, probably best to merge first and then optimize.

**Comment by dirkhh** [comment]
As I said, the latest Android Beta is based on this branch. So this is getting real world testing.
I played with it on half a dozen of my devices and tested downloading on a couple of them - I didn't notice any horrible hangs for recalculating profiles...

File: mobile-widgets/qml/main.qml [hunk 1]
```diff
@@ -552,8 +554,16 @@ if you have network connectivity and want to sync your data to cloud storage."),
 		property color contrastAccentColor: "#FF5722" // used for delete button
 		property color lightDrawerColor: "#FFFFFF"
 		property color darkDrawerColor: "#424242"
+		property int initialWidth: rootItem.width
 		Component.onCompleted: {
-			setupUnits()
+			// break the binding
+			initialWidth = initialWidth * 1
```
Comment (bstoeger) [inline mobile-widgets/qml/main.qml:562]: That is very obscure to someone not familiar with QML / JavaScript and might be worthy of a comment.
Is that a way to convert to int? How about `initialWidth + 0`?

File: subsurface-helper.cpp [hunk 2]
```diff
@@ -61,8 +61,20 @@ void exit_ui()
...
-
 #ifdef SUBSURFACE_MOBILE
+#if defined(Q_OS_ANDROID)
+	if (getAndroidHWInfo().contains("/OnePlus/")) {
+		QFontDatabase db;
+		int id = QFontDatabase::addApplicationFont(":/fonts/Roboto-Regular.ttf");
+		QString family = QFontDatabase::applicationFontFamilies(id).at(0);
+		QFont newDefaultFont;
+		newDefaultFont.setFamily(family);
+		(static_cast<QApplication *>(QCoreApplication::instance()))->setFont(newDefaultFont);
```
Comment (bstoeger) [inline subsurface-helper.cpp:73]: That cast is ugly. If I read the docs correctly, you can either user `QApplication::instance()` or `qApp` without cast.

**Review by bstoeger — APPROVED** [review]
Only two minor comments - let's merge and proceed...

File: mobile-widgets/qml/main.qml (hunk 1 above)
Comment (dirkhh) [reply mobile-widgets/qml/main.qml:562]: Actually, the comment is there. This is about breaking the binding in QML. Normally initialWidth would be "bound" to the value it got initialized with as a property - so whenever width changes, initialWidth would change. By making it the result of a computation, we are breaking this binding. I'm sure adding 0 would work the same.
https://doc.qt.io/qt-5/qtqml-syntax-propertybinding.html for more information

File: subsurface-helper.cpp (hunk 2 above)
Comment (dirkhh) [reply subsurface-helper.cpp:73]: stupidly we have a qApp macro that conflicts with the "normal" way of doing this :-(
Yes, the right solution would be to change that macro. I'll add fixing that to my long list and merge as is (as you suggested).

**Comment by dirkhh** [comment]
Thank you so much for taking the time to review, @bstoeger 
I always feel MUCH better if someone else takes another look.

File: mobile-widgets/qml/main.qml (hunk 1 above)
Comment (bstoeger) [reply mobile-widgets/qml/main.qml:562]: > Actually, the comment is there. This is about breaking the binding in QML. Normally initialWidth would be "bound" to the value it got initialized with as a property - so whenever width changes, initialWidth would change. By making it the result of a computation, we are breaking this binding. I'm sure adding 0 would work the same.

Whaat? I don't think I want to understand that. :-P

File: subsurface-helper.cpp (hunk 2 above)
Comment (bstoeger) [reply subsurface-helper.cpp:73]: > stupidly we have a qApp macro that conflicts with the "normal" way of doing this :-(

`git grep qApp` just shows a number of normal uses of `qApp`.

Comment (jaybny) [reply subsurface-helper.cpp:73]: is there a macro?

Comment (dirkhh) [reply subsurface-helper.cpp:73]: > is there a macro?

random question on an eight months old PR?

**Pull request merged**